test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		//Check we have dummy content.
+		const contentAfterFirstUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterFirstUndo );
+
+		// Click undo again, this time in the header. Reverts initial dummy content.
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		// Check dummy content is gone.
+		const contentAfterSecondUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterSecondUndo );
+	} );
+
+	test( 'should show the edited content after revert, clicking undo in the header toolbar, save and reload', async ( {
+		admin,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		await templateRevertUtils.save();
+
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the edited content after revert, clicking undo in the notice and reload', async ( {
+		admin,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		await templateRevertUtils.save();
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toBe( contentAfter );
+	} );
+} );
+
+class TemplateRevertUtils {
+	constructor( { editor, page } ) {
+		this.editor = editor;
+		this.page = page;
+	}
+
+	async addDummyText() {
+		await this.editor.insertBlock( { name: 'core/paragraph' } );
+		await this.page.keyboard.type( 'Test' );
+	}
+
+	async save() {
+		const headerToolbar = this.page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( 'role=button[name="Save"]' ).click();
+		// Second Save button in the entities panel.
+		const publishToolbar = this.page.locator(
+			'role=region[name="Publish"]'
+		);
+		await publishToolbar.locator( 'role=button[name="Save"]' ).click();
```

Comment:
Nit: We can omit the variable declarations since role selectors are already readable.

```suggestion
		await this.page.click(
			'role=region[name="Header"] >> role=button[name="Save"]'
		);
		// Second Save button in the entities panel.
		await this.page.click(
			'role=region[name="Publish"i] >> role=button[name="Save"i]'
		);
```

Reply:
We should probably also wait for the entities to be saved in this util. Probably a good idea to wait until the snackbar appears?